src/main/java/core/basesyntax/strategy/OperationStrategyImpl.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+
+    private final Map<FruitTransaction.Operation, OperationHandler> handlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> handlers) {
+        this.handlers = handlers;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        return handlers.get(operation);
```

getHandler(...) returns `handlers.get(operation)` directly which may return null if the operation is not present. The task requires that the strategy should throw an informative error when a handler is missing. This violates the requirement: "OperationStrategy returns the correct handler for a given transaction operation; throw informative error on missing handler." Instead, check for null and throw a RuntimeException with a clear message (also validate `operation` is not null). Example behaviour: retrieve handler into a variable, if null -> throw new RuntimeException("No handler for operation: " + operation).